common/lib/core-interfaces/test-d/index.test-d.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/* eslint-disable unicorn/filename-case */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {expectError} from "tsd";
+import { IFluidLoadable, IProvideFluidLoadable, Provider, ProviderKeys } from "../dist";
+
+
+declare function getUnknown(): Provider;
+
+declare function useUnknown(params: Provider | undefined): void;
+
+declare function useProvider<T extends Provider>(params: Provider<T> | undefined): void;
+
+declare function useProviderKey<T,TKey extends ProviderKeys<T> = ProviderKeys<T>>(key: TKey): void;
+
+declare function useLoadable(params: Provider<IFluidLoadable> | undefined): void;
+
+// test with provider
+{
+    const provider: Provider<IProvideFluidLoadable> = getUnknown();
+    useUnknown(provider);
+    useUnknown(provider.IFluidLoadable);
+    useProvider(provider);
+    useProvider(provider.IFluidLoadable);
+    useLoadable(provider);
+    useLoadable(provider.IFluidLoadable);
+    expectError(provider.handle);
+    provider.IFluidLoadable?.handle;
+    const unknown: Provider | undefined = provider.IFluidLoadable;
+    useUnknown(unknown);
+    useProvider(unknown);
+    useProvider<IFluidLoadable>(unknown);
+    useLoadable(unknown);
+}
+
+// test with interface
+{
+    const foo: Provider<IFluidLoadable> = getUnknown();
+    useUnknown(foo);
+    useUnknown(foo.IFluidLoadable);
+    useProvider(foo);
+    useProvider(foo.IFluidLoadable);
+    useLoadable(foo);
+    useLoadable(foo.IFluidLoadable);
+    expectError(foo.handle);
+    foo.IFluidLoadable?.handle;
+    const unknown: Provider | undefined = foo.IFluidLoadable;
```

Comment:
this is about testing implicit conversion between an unspecified, or unknown `Provider` and a concrete or known `Provider<IFoo>`. 

Think of the example of scope that the loader takes. It will take `scope: Provider` and i don't want to require a cast for a consumer to set `scope` to `Provider<IFoo>`. Similarly, within the runtime, when a consumer accesses scope i don't want to require a cast either, so want to support `const maybeFoo: Provider<IFoo> = contenxt.scope`

Reply:
i think you are just hung up on the naming here. nothing here has to do with the typescript type `unknown` as it is not useful in this context